recipe-server/client/control_new/components/data/QueryRecipes.js

```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { isEqual } from 'underscore';
+
+import { fetchFilteredRecipesPage } from 'control_new/state/recipes/actions';
+
+
+class QueryRecipes extends React.Component {
+  static propTypes = {
+    fetchFilteredRecipesPage: PropTypes.func,
+    filters: PropTypes.object,
+    pageNumber: PropTypes.number,
+  }
+
+  componentWillMount() {
+    const { filters, pageNumber } = this.props;
+    this.props.fetchFilteredRecipesPage(pageNumber, filters);
```

Comment:
The plucking works. It shadows the imported value, but it doesn't cause any problems. It is kind of a footgun though. Something we could do would be to change the import to `import * as recipeActions from 'control_new/state/recipes/actions'`, and change `connect` to reference is as `recipeActions.fetchFilteredRecipesPage`. It is a bit more verbose, and we lose some static analysis, but it avoids the shadowing problem. I think shadowing is a bigger concern.